src/js/jsx/tools/SuperselectOverlay.jsx

```diff
@@ -492,34 +492,38 @@ define(function (require, exports, module) {
                 }
             });
 
-            // Yuck, we gotta traverse the list backwards, and D3 doesn't offer reverse iteration
-            _.forEachRight(d3.selectAll(".layer-bounds")[0], function (element) {
-                var layer = d3.select(element),
-                    layerID = mathUtil.parseNumber(layer.attr("layer-id")),
-                    layerSelected = layer.attr("layer-selected") === "true",
-                    layerModel = layerTree.byID(layerID);
-
-                // Sometimes, the DOM elements may be out of date, and be of different documents
-                if (!layerModel) {
-                    return;
-                }
-                
-                var visibleBounds = layerTree.boundsWithinArtboard(layerModel),
-                    intersects = visibleBounds &&
-                        visibleBounds.left <= canvasMouse.x && visibleBounds.right >= canvasMouse.x &&
-                        visibleBounds.top <= canvasMouse.y && visibleBounds.bottom >= canvasMouse.y;
-
-                if (!underPolicy && !marquee && !highlightFound && intersects) {
-                    if (!layerSelected) {
-                        layer.classed("layer-bounds-hover", true)
-                            .style("stroke-width", 1.0 * scale);
-                    }
-                    highlightFound = true;
-                } else {
-                    layer.classed("layer-bounds-hover", true)
-                        .style("stroke-width", 0.0);
-                }
-            });
+            uiUtil.hitTestLayers(this.state.document.id, canvasMouse.x, canvasMouse.y)
+                .bind(this)
+                .then(function (hitLayerIDs) {
+                    return hitLayerIDs.findLast(function (id) {
+                        return !this.state.document.layers.byID(id).isArtboard;
+                    }, this);
+                })
+                .then(function (topID) {
+                    // Yuck, we gotta traverse the list backwards, and D3 doesn't offer reverse iteration
+                    _.forEachRight(d3.selectAll(".layer-bounds")[0], function (element) {
+                        var layer = d3.select(element),
+                            layerID = mathUtil.parseNumber(layer.attr("layer-id")),
+                            layerSelected = layer.attr("layer-selected") === "true",
+                            layerModel = layerTree.byID(layerID);
+
+                        // Sometimes, the DOM elements may be out of date, and be of different documents
+                        if (!layerModel) {
+                            return;
+                        }
+                        
+                        if (!underPolicy && !marquee && !highlightFound && layerID === topID) {
```

Comment:
I may be way off here, but now that we know about a definitive `topID` I have the vague feeling that we could avoid iterating over all the D3 elements, and instead do: "set all to no highlighted state" and then "grab just the topID element and apply some logic".  You don't even have to respond to this if I'm crazy.